great_expectations/data_context/config_validator/yaml_config_validator.py

```diff
@@ -0,0 +1,623 @@
+"""Config Validator for YAML based configs.
+
+This validator evaluates YAML configurations of core Great Expectations components to give feedback on
+ whether they have been configured correctly. It is linked to a Data Context since it does update the
+ configuration of the Data Context in some cases if the configuration is valid.
+
+ Typical usage example:
+ import great_expectations as ge
+ context = ge.get_context()
+ context.test_yaml_config(my_config)
+"""
+from __future__ import annotations
+
+import os
+import traceback
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Union, cast
+
+from ruamel.yaml import YAML
+from ruamel.yaml.comments import CommentedMap
+
+from great_expectations.checkpoint import Checkpoint, SimpleCheckpoint
+from great_expectations.core.usage_statistics.anonymizers.anonymizer import Anonymizer
+from great_expectations.core.usage_statistics.anonymizers.datasource_anonymizer import (
+    DatasourceAnonymizer,
+)
+from great_expectations.core.usage_statistics.usage_statistics import (
+    send_usage_message_from_handler,
+)
+from great_expectations.data_context.store import Store
+from great_expectations.data_context.types.base import (
+    CheckpointConfig,
+    DataContextConfig,
+    datasourceConfigSchema,
+)
+from great_expectations.data_context.util import (
+    instantiate_class_from_config,
+    substitute_all_config_variables,
+)
+from great_expectations.datasource import DataConnector, Datasource
+from great_expectations.rule_based_profiler import RuleBasedProfiler
+from great_expectations.rule_based_profiler.config import RuleBasedProfilerConfig
+from great_expectations.util import filter_properties_dict
+
+if TYPE_CHECKING:
+    from great_expectations.data_context import AbstractDataContext
+
+try:
+    from typing import Literal
+except ImportError:
+    # Fallback for python < 3.8
+    from typing_extensions import Literal  # type: ignore[misc]
+
+# TODO: check if this can be refactored to use YAMLHandler class
+yaml = YAML()
+yaml.indent(mapping=2, sequence=4, offset=2)
+yaml.default_flow_style = False
+
+
+class _YamlConfigValidator:
```

Comment:
Is this a convention we're using moving forward? I like it but what exactly does it mean for our users? That a piece of code is not to be touched/lacks extensibility?